core/service/manager.go

```diff
@@ -50,7 +49,14 @@ type DialogWaiterFactory func(providerID identity.Identity, serviceType string)
 type DialogHandlerFactory func(market.ServiceProposal, session.ConfigNegotiator) communication.DialogHandler
 
 // DiscoveryFactory initiates instance which is able announce service discoverability
-type DiscoveryFactory func() *discovery_registry.Discovery
+type DiscoveryFactory func() DiscoveryService
+
+// DiscoveryService registers the service to the discovery api periodically
+type DiscoveryService interface {
```

Comment:
Imho its ambiguous naming.. I would call it ServiceDiscovery, but since we are in a service package we could just call it 'Discovery'.